From cdaa6d08fff423e6244df17dbd7fd084a1ad1e6d Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sun, 4 Aug 2013 18:07:09 +0200
Subject: [PATCH 35/65] acerhk: drop autowlan module parameter

The autowlan module parameter is non-functional with dritek hardware.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 28 ----------------------------
 1 file changed, 28 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -77,21 +77,18 @@
 
 /* Module parameters */
 static int poll;
-static int autowlan;
 static int wlan_state = -1;
 static int bluetooth_state = -1;
 static int verbose = 1;
 static unsigned int force_series;
 
 module_param(poll, int, 0444);
-module_param(autowlan, int, 0444);
 module_param(verbose, int, 0444);
 module_param(wlan_state, int, 0444);
 module_param(bluetooth_state, int, 0444);
 module_param(force_series, uint, 0444);
 
 MODULE_PARM_DESC(poll, "start polling timer");
-MODULE_PARM_DESC(autowlan, "automatic switching of wlan hardware");
 MODULE_PARM_DESC(wlan_state, "(assumed) initial state of WLAN LED/hardware");
 MODULE_PARM_DESC(bluetooth_state,
 		 "(assumed) initial state of Bluetooth LED/hardware");
@@ -1062,24 +1059,6 @@ static t_key_names transl8_key_code(int
 		else if (keyname == k_none && verbose > 3)
 			pr_info("translated acer key code 0x%x to no key\n",
 				keycode);
-
-		if (autowlan) {
-			/* if automatic switching of wlan hardware is enabled,
-			 * do it here on wireless key press */
-			if (keyname == k_wireless2) {
-				if (acerhk_bluetooth_state)
-					wbutton_fct_1(0);
-				else
-					wbutton_fct_1(1);
-			}
-
-			if (keyname == k_wireless) {
-				if (acerhk_wlan_state)
-					wbutton_fct_2(0);
-				else
-					wbutton_fct_2(1);
-			}
-		}
 	}
 
 	return keyname;
@@ -1635,13 +1614,6 @@ static int acerhk_probe(struct platform_
 		pr_info("Acer Travelmate hotkey driver loaded, configured laptop series: %d\n",
 			acerhk_series);
 
-		/* If automatic switching of wlan is wanted but polling is
-		 * disabled, automatically enable it */
-		if (!poll && autowlan) {
-			pr_info("Automatic switching of wireless hardware needs polling, enabling it\n");
-			poll = 1;
-		}
-
 		/* start automatic polling of key presses if wanted and bios
 		 * routine found */
 		if (poll && bios_routine)
